src/lightning/data/streaming/dataset.py

```diff
@@ -158,10 +188,10 @@ def __next__(self) -> Any:
         return data
 
 
-def _try_create_cache_dir(input_dir: str) -> Optional[str]:
+def _try_create_cache_dir(input_dir: str, shard_rank: int = 0) -> Optional[str]:
     if "LIGHTNING_CLUSTER_ID" not in os.environ or "LIGHTNING_CLOUD_PROJECT_ID" not in os.environ:
         return None
     hash_object = hashlib.md5(input_dir.encode())
-    cache_dir = os.path.join(f"/cache/chunks/{hash_object.hexdigest()}")
+    cache_dir = os.path.join("/cache", hash_object.hexdigest(), str(shard_rank), "chunks")
```

Comment:
Which format do you prefer?
1) `/cache/chunks/hash/worker_id
`
or
2) `/cache/hash/worker_id/chunks
`

Reply:
I prefer 1) This makes more sense as the hash is the `input_dir`.